devdoc/doublylinkedlist_requirements.md

```diff
@@ -104,3 +127,54 @@ extern PDLIST_ENTRY DList_RemoveHeadList(PDLIST_ENTRY listHead);
 **SRS_DLIST_06_013: [** DList_RemoveHeadList shall return listHead if that's the only item in the list. **]**
 
 Note: The Flink & Blink of the returned PDLIST_ENTRY shall be undefined.
+
+### DList_Find
+```c
+MOCKABLE_FUNCTION(, int, DList_Find, PDLIST_ENTRY, listHead, DLIST_MATCH_FUNCTION, matchFunction, const void*, matchContext, PDLIST_ENTRY*, foundEntry);
+```
+
+**SRS_DLIST_43_001: [** `DList_Find` shall call `matchFunction` on each entry in the list defined by `listHead` along with `matchContext`. **]**
+
+**SRS_DLIST_43_002: [** If the call to `matchFunction` for an entry returns `DLIST_MATCH_FUNCTION_MATCHING`, `DList_Find` shall set `foundEntry` to that entry. **]**
+
+**SRS_DLIST_43_003: [** If no calls to `matchFunction` return `DLIST_MATCH_FUNCTION_MATCHING`, `DList_Find` shall set `foundEntry` to `NULL`. **]**
+
+**SRS_DLIST_43_015: [** `DList_Find` shall succeed and return zero. **]**
+
+**SRS_DLIST_43_016: [** If there are any failures, `DList_Find` shall fail and return a non-zero value. **]**
+
+### DList_RemoveIf
+```c
+MOCKABLE_FUNCTION(, int, DList_RemoveIf, PDLIST_ENTRY, listHead, DLIST_CONDITION_FUNCTION, conditionFunction, const void*, conditionContext, DLIST_ENTRY_DESTROY_FUNCTION, destroyFunction, const void*, destroyContext, PDLIST_ENTRY*, newHead);
```

Comment:
>newHead [](http://example.com/codeflow?start=232&length=7)

I'm not sure why we need this?

Reply:
I can't see right now a use case for this, but maybe I am blind. Can you clarify?

Reply:
Still waiting for this.

Reply:
The head of the list could be removed. The user will need a pointer to the new head of the list.

Reply:
Can you write me the user code for that scenario please?

Reply:
Tried to come up with one. In the process, I understood why it won't be needed. Removing.